tests/integration/Payment/UseCases/PaymentCompletedEmailTest.php

```diff
@@ -98,86 +94,58 @@ public function testWhenGroupHasNoMailCredentialsSetNothingHappens() : void
         $this->paymentService->completePayment(1);
 
         $this->assertPaymentWasCompleted();
-        $this->tester->seeEmailCount(0);
     }
 
     public function testEmailIsSentWhenPaymentIsCompleted() : void
     {
-        $email = new EmailTemplate('subject', 'body');
-        $this->createMailCredentials();
-        $this->initEntities([
-            EmailType::PAYMENT_INFO => new EmailTemplate('', ''),
-            EmailType::PAYMENT_COMPLETED => $email,
-        ]);
-
-        $this->users->setUser(new User(1, 'František Maša', 'frantisekmasa1@gmail.com'));
-
-        $this->paymentService->completePayment(1);
-
-        $this->assertPaymentWasCompleted();
-        $this->tester->seeEmailCount(1);
-        $this->tester->seeInLastEmailSubjectTo(self::EMAIL, $email->getSubject());
-        $this->tester->seeInLastEmailTo(self::EMAIL, $email->getBody());
-    }
-
-    public function testWhenEmailCannotBeSentViaSmtpPaymentIsCompletedAndUserIsNotified() : void
```

Comment:
Tenhle test taky zmizel a měl by být "jen" předělán na nový kód.

Reply:
taky nevim jak to testovat

Reply:
Asi bych šel jednodušší cestou a předával implementaci `IMailer`, která vyhodí `SendException`. Ta by měla být odchytávána v MailingService stejně jako byla předtím odchytávána `SmtpException`.

https://github.com/skaut/Skautske-hospodareni/blob/7bfa911ac0ca8e12abfc53f9c9ca88035f43ec45/app/model/Payment/MailingService.php#L188

Stejně tak koukám, že v `OAuthMailer` chybí nějaký error handling. Komunikace s Google může vyhodit `Google_Exception`. Tu by měl email odchytit a vyhodit zabalenou v `SendException`.

Ty testy tam nejsou jen tak pro srandu :slightly_smiling_face: Tenhle testuje, že pokud se podaří spárovat platbu, ale nikoliv odeslat email, tak se uživatel dozví, že se tak stalo. Momentálně by dostal pětikilo.

Reply:
Udělal jsem z toho novy test `PaymentCompletedEmailFailTest.php` protoze jsem potreboval nahradit NullMailer za takovy, ktery hodi chybu.